File: backend/app/Enums/ReportCategory.php
<?php

namespace App\Enums;

use Filament\Support\Contracts\HasLabel;

/**
 * reports.category. docs/07-ui-ux-design.md §3.7 "Report — category" lists
 * these exact six as the report form's options, sourced from
 * `GET /safety/report-categories`.
 */
enum ReportCategory: string implements HasLabel
{
    case Harassment = 'harassment';
    case FakeProfile = 'fake_profile';
    case Spam = 'spam';
    case InappropriateContent = 'inappropriate_content';
    case Scam = 'scam';
    case Other = 'other';

    public function getLabel(): string
    {
        return match ($this) {
            self::Harassment => 'Harassment',
            self::FakeProfile => 'Fake profile',
            self::Spam => 'Spam',
            self::InappropriateContent => 'Inappropriate content',
            self::Scam => 'Scam',
            self::Other => 'Other',
        };
    }
}

File: mobile/lib/safety/domain/report_category.dart
/// Mirrors `App\Enums\ReportCategory` on the backend. Hardcoded here rather
/// than fetched from `GET /safety/report-categories` — the same
/// client-mirrors-the-backend-enum convention as `SwipeDirection`/
/// `MessageType`, not a dynamically-loaded list. docs/07-ui-ux-design.md
/// §3.7 "Report — category" lists these exact six labels.
enum ReportCategory {
  harassment('harassment', 'Harassment'),
  fakeProfile('fake_profile', 'Fake profile'),
  spam('spam', 'Spam'),
  inappropriateContent('inappropriate_content', 'Inappropriate content'),
  scam('scam', 'Scam'),
  other('other', 'Other');

  const ReportCategory(this.apiValue, this.label);

  final String apiValue;
  final String label;
}
